src/ui.tsx

```diff
@@ -589,6 +589,39 @@ function PlanStatistics({plan}: {plan: Plan}) {
   </div>;
 }
 
+function TimeLeft({plan, progress, currentLineStartedTime, paused}: {
+  plan: Plan;
+  progress: number | null; 
+  currentLineStartedTime: Date | null;
+  paused: boolean;
+}) {
+  const [_, setTime] = useState(new Date());
+
+  // Interval that ticks every second to rerender
+  // and recalculate time remaining for long lines
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setTime(new Date());
+    }, 1000);
+
+    () => {
+      clearInterval(interval);
+    }
+  }, [setTime])
+
+  if (!plan || !plan.duration || progress === null || paused) {
+    return null;
+  }
+
+  const currentLineProgress = (new Date().getTime() - currentLineStartedTime.getTime()) / 1000;
```

Comment:
Also here; use `motion` instead of `line`.

Maybe also `TimeRemaining` instead of `Progress`?

```suggestion
  const currentMotionTimeRemaining = (new Date().getTime() - currentMotionStartedTime.getTime()) / 1000;
```

Reply:
It is not "time remaining", but the "time spent". I'll rename it to `currentMotionTimeSpent` for now, but I'm open for suggestions.

Reply:
Ah, okay. "time spent" seems okay!